### PR TITLE
Do not treat CompileOptions.incremental as input

### PR DESCRIPTION
**Pull request by lptr**
Changing this parameter should not influence the output of the task in any way, and thus should not be treated as an input.

### CODE REVIEW

**Comment by lptr** [comment]
Build: https://builds.gradle.org/viewQueued.html?itemId=10463484

File: subprojects/plugins/src/integTest/groovy/org/gradle/java/compile/IncrementalCompileMultiProjectTestFixture.groovy [hunk 1]
```diff
@@ -42,15 +43,15 @@ trait IncrementalCompileMultiProjectTestFixture {
         file('app/src/main/java/AClass.java') << 'public class AClass { }'
     }
 
-    def getAppCompileJava() {
+    static String getAppCompileJava() {
```
Comment (wolfs) [inline subprojects/plugins/src/integTest/groovy/org/gradle/java/compile/IncrementalCompileMultiProjectTestFixture.groovy]: I think we shouldn't use static methods on `trait`s: http://docs.groovy-lang.org/next/html/documentation/core-traits.html#_static_methods_properties_and_fields

File: subprojects/plugins/src/integTest/groovy/org/gradle/java/compile/IncrementalJavaCompileIntegrationTest.groovy [hunk 2]
```diff
@@ -123,6 +124,41 @@ class IncrementalJavaCompileIntegrationTest extends AbstractIntegrationSpec impl
         executedAndNotSkipped(libraryCompileJava)
     }
 
+    @Unroll
+    def "does not recompile when only compileOptions.incremental property changes from #from to #to"() {
```
Comment (wolfs) [inline subprojects/plugins/src/integTest/groovy/org/gradle/java/compile/IncrementalJavaCompileIntegrationTest.groovy]: I am not so sure where to place this test. It is actually not really related to incremental Java compile - it is actually related to incremental build. Maybe `JavaCompileIntegrationTest` is the way to go?

File: subprojects/plugins/src/integTest/groovy/org/gradle/java/compile/IncrementalCompileMultiProjectTestFixture.groovy (hunk 1 above)
Comment (lptr) [reply subprojects/plugins/src/integTest/groovy/org/gradle/java/compile/IncrementalCompileMultiProjectTestFixture.groovy]: Good point, removed the `static` modifiers.

File: subprojects/plugins/src/integTest/groovy/org/gradle/java/compile/IncrementalJavaCompileIntegrationTest.groovy (hunk 2 above)
Comment (lptr) [reply subprojects/plugins/src/integTest/groovy/org/gradle/java/compile/IncrementalJavaCompileIntegrationTest.groovy]: It _is_ about enabling/disabling the incremental compiler, so I'd leave it here.

**Pull request merged**